runtime/jvmti/jvmtiHook.c

```diff
@@ -1056,18 +1061,19 @@ jvmtiHookSingleStep(J9HookInterface** hook, UDATA eventNum, void* eventData, voi
 	J9VMSingleStepEvent * data = eventData;
 	J9JVMTIEnv * j9env = userData;
 	jvmtiEventSingleStep callback = j9env->callbacks.SingleStep;
+	J9VMThread * currentThread = data->currentThread;
 
 	Trc_JVMTI_jvmtiHookSingleStep_Entry();
 
 	ENSURE_EVENT_PHASE_LIVE(jvmtiHookSingleStep, j9env);
 
 	/* Call the event callback */
-	if (callback != NULL) {
-		J9VMThread * currentThread = data->currentThread;
+	if (callback != NULL && shouldPostEvent(currentThread, NULL)) {
 		jthread threadRef;
 		UDATA hadVMAccess;
 		UDATA javaOffloadOldState = 0;
 
+
```

Comment:
Empty line can be removed.

Reply:
Please remove this extra, unwanted blank line as Babneet requested.